### ISSUE TITLE
OnICECandidate emit another candidate after a nil

### ISSUE THREAD

**Issue by enobufs** [pion/webrtc#1138]
### Your environment.
  - Version: v2.2.0 and current HEAD (f4f8299)

### What did you do?
Just connecting to another pion node on the same local interface.

### What did you expect?
The last OnICECandidate callback to be reporting nil.

### What happened?
OnICECandidate reports one more non-nil candidate after reporting the nil.

```
session DEBUG: 00:08:30.900923 session.go:1008: onICECandidate: host 10.0.0.135:55186
session DEBUG: 00:08:30.955851 session.go:1008: onICECandidate: srflx 98.207.205.254:52317 related 0.0.0.0:52317
session DEBUG: 00:08:31.012624 session.go:1008: onICECandidate: <nil>
session DEBUG: 00:08:31.012640 session.go:1008: onICECandidate: srflx 2601:646:c202:12d0:252d:b3c8:8292:d0be:52318 related :::52318
session DEBUG: 00:08:31.012774 session.go:1018: onICECandidate: SDP:
{"type":"offer","sdp":"v=0\r\no=- 749561647 1586675310 IN IP4 0.0.0.0\r\ns=-\r\nt=0 0\r\na=fingerprint:sha-256 55:34:DD:A1:AC:47:C0:AE:28:05:6C:4A:D5:D4:29:6E:45:1C:E8:70:4B:62:5C:16:41:0B:49:7C:6B:1C:65:B4\r\na=group:BUNDLE 0\r\nm=application 9 DTLS/SCTP 5000\r\nc=IN IP4 0.0.0.0\r\na=setup:actpass\r\na=mid:0\r\na=sendrecv\r\na=sctpmap:5000 webrtc-datachannel 1024\r\na=ice-ufrag:SRNkuKaHuQKnRtPj\r\na=ice-pwd:lXyLkYJWDyomXUAEFrgqKwRnyIHoJLKX\r\na=candidate:foundation 1 udp 2130706431 10.0.0.135 55186 typ host generation 0\r\na=candidate:foundation 2 udp 2130706431 10.0.0.135 55186 typ host generation 0\r\na=candidate:foundation 1 udp 1694498815 98.207.205.254 52317 typ srflx raddr 0.0.0.0 rport 52317 generation 0\r\na=candidate:foundation 2 udp 1694498815 98.207.205.254 52317 typ srflx raddr 0.0.0.0 rport 52317 generation 0\r\na=candidate:foundation 1 udp 1694498815 2601:646:c202:12d0:252d:b3c8:8292:d0be 52318 typ srflx raddr :: rport 52318 generation 0\r\na=candidate:foundation 2 udp 1694498815 2601:646:c202:12d0:252d:b3c8:8292:d0be 52318 typ srflx raddr :: rport 52318 generation 0\r\na=end-of-candidates\r\n"}
```

It appears that pc.LocalDescription contains the candidate reported after the nil, so if the app reference the local description after receiving the nil (a typical case) won't be affected, but other cases may miss the last candidate.

Other notes:
* Trickle ICE is enabled.

**Comment by Sean-Der** [comment]
https://github.com/pion/ice/pull/168 we did land a fix to try and fix this!

The root issue is that goroutines may not fire in the order they were created, so we fire them in a queue now. Somewhere in the stack we might still be doing this. I can look at this soon, just doing H264 RTP Unmarshal tonight :)

**Comment by enobufs** [comment]
@Sean-Der OK - Great!

**Comment by enobufs** [comment]
The PR pion/ice#168 appears to have been landed on its master as v0.7.12. Which lannded on pion/webrtc v2.2.5. Closing this issue.
